validator/app.py

```
@@ -2,751 +2,64 @@
-    pages = data.groupby("page_vuid")
-    data_json = [
-        {
-            "page_vuid": page[0],
-            "questions": json.loads(
-                page[1][["exercise_uid", "option_words", "stem_words"]].to_json(
-                    orient="records"
-                )
-            ),
-        }
-        for page in pages
-    ]
-    return jsonify(data_json)
-
-
-@app.route("/datasets/books/<vuid>/vocabularies/questions/<pvuid>")
-def fetch_page_questions(vuid, pvuid):
-    data = df_questions[df_questions["cvuid"] == ":".join((vuid, pvuid))].rename(
-        {"uid": "exercise_uid", "mc_words": "option_words"}, axis=1
-    )
-    if data.empty:
-        _validate_vuid(vuid)
-        _validate_vuid(pvuid, vuid_type="page")
-        book = df_domain[df_domain["vuid"] == vuid]
-        if book.empty:
-            raise InvalidUsage("No such book", status_code=404)
-        page = df_innovation[df_innovation["cvuid"] == ":".join((vuid, pvuid))]
-        if page.empty:
-            raise InvalidUsage("No such page in book", status_code=404)
-
-        return jsonify([])
-
-    json_data = json.loads(
-        data[["exercise_uid", "option_words", "stem_words"]].to_json(orient="records")
-    )
-    return jsonify(json_data)
-
-
-@app.route("/datasets/questions")
-def questions_index():
-    return jsonify(df_questions.uid.tolist())
-
-
-@app.route("/datasets/questions/<uid>")
-def fetch_question(uid):
-    data = df_questions[df_questions["uid"] == uid].rename(
-        {"uid": "exercise_uid", "mc_words": "option_words"}, axis=1
-    )
-
-    json_data = json.loads(
-        data[["exercise_uid", "option_words", "stem_words"]].to_json(orient="records")
-    )
-    return jsonify(json_data)
-
+        os.listdir(data_dir)
+    except FileNotFoundError:
+        data_dir = tempfile.mkdtemp()
+        app.config["DATA_DIR"] = data_dir
 
-@app.route("/ping")
-def ping():
-    return "pong"
+        def del_temp_data(dd=data_dir):
```

Guess we only support 1 process per server? Otherwise stuff might get cleaned up when 1 process exits. Also if `DATA_DIR` is not set this might clean... everything including the app dir?

---

Hmm, multiprocess _might_ be an issue, I'll test that. As to not set, this def only exists under a `FIleNotFound` exception catch. That fires for `os.listdir('')` (which is why that default exists on the `app.config.get()`  - `os.listdir(None)` gives you the cwd!) I'll see what happens w/ launching under gunicorn w/ multiple workers, and kill one of the children. For now, in fact, I found a problem with multiprocess for importing books, so we currently only support single-process, mulit-thread, for this release.

---

Hehe, just tested - guess what? Each worker process gets its _own_ data_dir! Not exactly the best solution. I think I should probably remove this code and just error out if data_dir is not set.